### PR TITLE
Feature/voice messages

### PR DESCRIPTION
**Pull request by hamzahass**
Re-allowed requesting permissions for audio and external storage on local emulator, which fixed bugs for cirrius.

### CODE REVIEW

**Review by ha2san — APPROVED** [review]
Nice, I see that you found a way to grant the permission. I think we should try to add some dependencies to be able to test everything. Other than that, everything seems to be ok

File: app/build.gradle [hunk 1]
```diff
@@ -60,6 +60,13 @@ dependencies {
     def lifecycle_version = "2.5.0-alpha05"
     def arch_version = "2.1.0"
 
+    //for AppCompat use:
```
Comment (DeepGreen1) [inline app/build.gradle:63]: Are those comments needed?

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt [hunk 2]
```diff
@@ -41,11 +55,33 @@ class ChatActivity : AppCompatActivity() {
...
 
+        buttonSendTextMessage = findViewById(R.id.buttonSendMessage)
+
         initMessageList(model)
         initSendMessageButton(model)
         initSendLocationButton(model, locationManager, fusedLocationClient)
         initSendImageCameraButton(model)
         initSendImageButton(model)
+
+        //initSendImageCameraButton() // TODO: Implement this
```
Comment (DeepGreen1) [inline app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:66]: I think that this method is implemented, why did you comment this line? If it stays like that, we will not able to send images using the camera.

**Review by DeepGreen1 — CHANGES_REQUESTED** [review]
Nice work! I think that we do the same for other parts that uses permissions, It'll improve our code coverage.
I let you some comments and I'll let you see for the takeCameraImage function that is commented. Otherwise everything seems fine to me. Great job!

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt (hunk 2 above)
Comment (hamzahass) [reply app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:66]: I don't understand this, as I did not comment this line, nor add the TODO, I will investigate on that ..

File: app/build.gradle (hunk 1 above)
Comment (hamzahass) [reply app/build.gradle:63]: Yes, I will need this for the view of recording messages

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt (hunk 2 above)
Comment (hamzahass) [reply app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:66]: After further investigation, this function call is made 3 lines above, and this TODO was surely added on my side when merging.

File: app/build.gradle (hunk 1 above)
Comment (DeepGreen1) [reply app/build.gradle:63]: Alright! We keep it then ;)

File: app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt (hunk 2 above)
Comment (DeepGreen1) [reply app/src/main/java/com/github/brugapp/brug/ui/ChatActivity.kt:66]: No problem this was probably overlooked, we'll have to remove it anyway. Let's do it during next print as the ChatActivity will again be modified during next week.

**Comment by DeepGreen1** [comment]
Alright, thanks for your answers. Everything seems fine to me!

**Review by DeepGreen1 — APPROVED** [review]

**Pull request merged**